src/components/HeaderAuthOptions/HeaderSignUp/HeaderSignUp.js

```diff
@@ -15,20 +31,80 @@ class HeaderSignUp extends Component {
 
     this.state = {
       email: '',
+      openEmal: false,
+      openPass: false,
+      openUser: false,
+      emailAvail: false,
+      emailValid: false,
+      emailVal: false,
       username: '',
+      usernameVal: false,
+      usernameLength: false,
+      usernameTaken: false,
       identifier: '',
+      submit: false,
       password: '',
-      passwordConfirmation: ''
+      passwordVal: false,
+      passwordLength: false,
+      passwordCapital: false,
+      passwordLower: false,
+      passwordSpecial: false,
+      passwordNumber: false,
+      passwordConfirmation: '',
+      passwordConfirmationVal: false,
+      property: document.documentElement.style.setProperty('--border-show', 'none')
     }
   }
+  // Set state of all fields using helper functions
+  checkValid = () => {
+    this.setState({ emailVal: emailTest(this.state.email, this.state.emailAvail) })
+    this.setState({ emailValid: emailValid(this.state.email) })
+    this.setState({ usernameVal: usernameTest(this.state.username, this.state.usernameTaken) })
+    this.setState({ usernameLength: usernameLength(this.state.username) })
+    this.setState({ passwordVal: passwordTest(this.state.password) })
+    this.setState({ passwordLength: passwordLength(this.state.password) })
+    this.setState({ passwordCapital: passwordCapital(this.state.password) })
+    this.setState({ passwordLower: passwordLower(this.state.password) })
+    this.setState({ passwordSpecial: passwordSpecial(this.state.password) })
+    this.setState({ passwordNumber: passwordNumber(this.state.password) })
+    this.setState({ passwordConfirmationVal: passwordConfirmationTest(this.state.password, this.state.passwordConfirmation) })
+  }
 
-  handleChange = event => this.setState({
-    [event.target.name]: event.target.value
-  })
+  handleChange = event => {
```

Comment:
This code is difficult to follow and somewhat repetitive. How can we refactor? What are we trying to accomplish with this function?

Reply:
I wanted the ability to check username and email while the user is typing. I only need to do this on two of the four form fields. The other two form fields still need the checkValid function. I went with the conditional to get this to work, but I would love to clean it up some.

The two api calls for checkname() and checkemail() are what drove me to do it this way originally. I may be able to rename the functions and condense down one of the conditionals, but I'll have to play with some ideas. I'm open to suggestions.

Reply:
I refactored the api calls so we can send two arguments. This allowed for the refactoring of handleChange a bit. 

Maybe we can still do something with `() => { this.checkValid()`? I was toying with the idea of placing it once at the end of the function, but I'm having trouble thinking of a way to call the api only when necessary.